cime_config/acme/machines/config_machines.xml

```diff
@@ -993,33 +993,100 @@
 	   <arguments>
-	     <arg name="mpinone"> --mpi=none</arg>
-	     <arg name="num_tasks"> --ntasks=$TOTALPES</arg>
-	     <arg name="cpubind"> --cpu_bind=sockets</arg>
-	     <arg name="cpubind"> --cpu_bind=verbose</arg>
-             <arg name="killonbadexit"> --kill-on-bad-exit</arg>
+	     <arg name="mpi">--mpi=none</arg>
+	     <arg name="num_tasks">--ntasks=$TOTALPES</arg>
+	     <arg name="cpu_bind">--cpu_bind=sockets --cpu_bind=verbose</arg>
+	     <arg name="kill-on-bad-exit">--kill-on-bad-exit</arg>
 	   </arguments>
 	 </mpirun>
-	 <mpirun mpilib="mpi-serial">
-           <executable></executable>
+	 <mpirun mpilib="mvapich">
+	   <executable>srun</executable>
+	   <arguments>
+	     <arg name="num_tasks">--ntasks=$TOTALPES</arg>
+	     <arg name="cpu_bind">--cpu_bind=sockets --cpu_bind=verbose</arg>
+	     <arg name="kill-on-bad-exit">--kill-on-bad-exit</arg>
+	   </arguments>
 	 </mpirun>
-
+	 <mpirun mpilib="intelmpi">
+	   <executable>mpirun</executable>
+	   <arguments>
+	     <arg name="num_tasks">-n $TOTALPES</arg>
+	   </arguments>
+	 </mpirun>
+	 <mpirun mpilib="openmpi">
+	   <executable>mpirun</executable>
+	   <arguments>
+	     <arg name="num_tasks">-n $TOTALPES</arg>
+	   </arguments>
+	 </mpirun>
+	 <module_system type="module">
+	   <init_path lang="perl">/share/apps/modules/Modules/3.2.10/init/perl.pm</init_path>
+	   <init_path lang="python">/share/apps/modules/Modules/3.2.10/init/python.py</init_path>
+	   <init_path lang="csh">/etc/profile.d/modules.csh</init_path>
+	   <init_path lang="sh">/etc/profile.d/modules.sh</init_path>
+	   <cmd_path lang="perl">/share/apps/modules/Modules/3.2.10/bin/modulecmd perl </cmd_path>
+	   <cmd_path lang="python">/share/apps/modules/Modules/3.2.10/bin/modulecmd python</cmd_path>
+	   <cmd_path lang="sh">module</cmd_path>
+	   <cmd_path lang="csh">module</cmd_path>
+	   <modules>
+	     <command name="purge"></command>
+	   </modules>
+	   <modules>
+	     <command name="load">perl/5.20.0</command>
+	     <command name="load">cmake/3.3.0</command>
+	     <command name="load">python/2.7.8</command>
+	   </modules>
+	   <modules compiler="intel">
+	     <command name="load">intel/15.0.1</command>
+	     <command name="load">mkl/15.0.1</command>
+	   </modules>
+	   <modules compiler="pgi">
+	     <command name="load">pgi/14.10</command>
+	   </modules>
+	   <modules mpilib="mvapich">
+	     <command name="load">mvapich2/2.1</command>
+	   </modules>
+	   <modules mpilib="mvapich2">
+	     <command name="load">mvapich2/2.1</command>
+	   </modules>
+	   <modules mpilib="intelmpi">
+	     <command name="load">intelmpi/5.0.1.035</command>
+	   </modules>
+	   <modules mpilib="openmpi">
+	     <command name="load">openmpi/1.8.3</command>
+	   </modules>
+	   <modules>
+	     <command name="load">netcdf/4.3.2</command>
+	   </modules>
+	 </module_system>
+	 <environment_variables>
+	   <env name="OMP_STACKSIZE">64M</env>
+	   <env name="NETCDF_HOME">$NETCDF_LIB/../</env>
```

Comment:
@jgfouca : If I use $ENV{NETCDF_LIB}, the makefile picks up `-I{NETCDF_LIB}/..//include` as include path. For some reason, $ENV is not expanding NETCDF_LIB variable.  Following are the contents of my .env_mach_specific.sh:
module purge
module load perl/5.20.0
module load cmake/3.3.0
module load python/2.7.8
module load intel/15.0.1
module load mkl/15.0.1
module load mvapich2/2.1
module load netcdf/4.3.2
export OMP_STACKSIZE=64M
export NETCDF_HOME=$ENV{NETCDF_LIB}/../
export MKL_PATH=$ENV{MLIB_LIB}


I am a csh user, contents of my .env_mach_specific.csh are:

module purge
module load perl/5.20.0
module load cmake/3.3.0
module load python/2.7.8
module load intel/15.0.1
module load mkl/15.0.1
module load mvapich2/2.1
module load netcdf/4.3.2
setenv OMP_STACKSIZE 64M
setenv NETCDF_HOME $ENV{NETCDF_LIB}/../
setenv MKL_PATH $ENV{MLIB_LIB}